BASE/src/Microsoft.ApplicationInsights/Channel/TransmissionStatusEventArgs.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace Microsoft.ApplicationInsights.Channel
+{
+    using System;
+    using System.Net.Http;
+
+    /// <summary>
+    /// Event argument to track response from ingestion endpoint.
+    /// </summary>
+    public class TransmissionStatusEventArgs : EventArgs
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TransmissionStatusEventArgs"/> class.
+        /// </summary>
+        /// <param name="response">Response from ingestion endpoint.</param>
+        public TransmissionStatusEventArgs(HttpResponseMessage response = null)
+        {
+            this.Response = response;
+        }
```

Comment:
Why do you need this new class as a wrapper around the HttpResponseMessage?
This is noteworthy to me because this will be a lot of new allocations.

Reply:
EventHandler requires<TEventArgs>. TransmissionStatusEventArgs class inherits EventArgs and contains HttpresponseMessage. I agree, there will be new allocation only if subscriber is listening to it. As you said this is a wrapper and hold only reference to HttpResponseMessage. When HttpResponseMessage is disposed in Transmission, Response property in this object also goes null.

In reality, this object is created and destroyed immediately as the event handler code runs synchronously.